src/detect-engine.c

```diff
@@ -958,7 +958,20 @@ static uint32_t DetectBufferTypeHashNameFunc(HashListTable *ht, void *data, uint
     const DetectBufferType *map = (DetectBufferType *)data;
     uint32_t hash = hashlittle_safe(map->name, strlen(map->name), 0);
     hash += hashlittle_safe((uint8_t *)&map->transforms, sizeof(map->transforms), 0);
+    // Add transform data, if any. Note that the transform identifier is always added; if
+    // the transform supports serialization, that data will also be added.
+    for (int i = 0; i < map->transforms.cnt; i++) {
+        const TransformData *t = &map->transforms.transforms[i];
+        hash += hashlittle_safe((uint8_t *)&t->transform, sizeof(t->transform), 0 );
+        if (sigmatch_table[t->transform].TransformSerialize) {
```

Comment:
I would want this to run in the caller instead.

Reply:
compare func also needs to be updated

Reply:
This line checks if there's a `TransformSerialize` function and if so, it's invoked in the next step. What part would/should move to the caller?

Reply:
> compare func also needs to be updated

Will do.

Reply:
the hash function shouldn't modify any data, it should just calc a hash. So prep work needs to be done before the hash func is called